net/httpclient.go

```diff
@@ -75,6 +76,10 @@ func NewClient(o Options) *Client {
 	return c
 }
 
+func (c *Client) SetEtag(etag string) {
```

Comment:
This assumes we only use one URL per client, but you could also have clients that query API x and API y that both support ETag. If you introduce a setter, maybe also a getter and pass hostname?

Reply:
My thinking was wrong here, because we have one remote for one dataclient.